src/textual/reactive.py

```diff
@@ -301,16 +273,18 @@ def _compute(cls, obj: Reactable) -> None:
             obj: Reactable object.
         """
         _rich_traceback_guard = True
-        for compute in obj._reactives.keys():
+        for name, reactive in obj._reactives.items():
+            if not reactive._run_compute:
+                continue
             try:
-                compute_method = getattr(obj, f"compute_{compute}")
+                compute_method = getattr(obj, f"compute_{name}")
             except AttributeError:
                 continue
-            current_value = getattr(obj, f"_reactive_{compute}")
+            current_value = getattr(obj, f"_reactive_{name}")
             value = compute_method()
-            setattr(obj, f"_reactive_{compute}", value)
-            if value != current_value:
-                cls._check_watchers(obj, compute, current_value)
+            setattr(obj, f"_reactive_{name}", value)
```

Comment:
one final comment for now: if all my guesses / proposals make sense, then several places that are currently class methods and take `obj`, can be converted to instance methods this way (with iterating over `_reactives`). *This would mean* that they can use `self.internal_name`, e.g., instead of duplicating that `f"_reactive_{name}"` bit of code over and code.

Reply:
Maybe for the next PR :)